### PR TITLE
🎉 Source Zendesk Talk: add Incremental sync

### LINKED ISSUE #2734: Source Zendesk Talk: Update source with incremental sync method
## Tell us about the problem you're trying to solve
Update the connector code for support the Incremental streams.

_Sub-task for Issue_ #2346

### PR DESCRIPTION
**Pull request by yevhenii-ldv**
## What
Added support of Incremental sync for Zendesk Talk connector.

closes #2734 

## Pre-merge Checklist
- [ ] *Run integration tests*
- [ ] *Publish Docker images*

### CODE REVIEW

**Review by keu — APPROVED** [review]

File: airbyte-integrations/connectors/source-zendesk-talk/source_zendesk_talk/api.py [hunk 1]
```diff
@@ -148,6 +148,49 @@ def read(self, getter: Callable) -> Iterator:
...
+    def state(self) -> Optional[Mapping[str, Any]]:
+        """Current state, if wasn't set return None"""
+        if self._state:
+            return {self.state_pk: str(self._state)}
+        return None
+
+    @state.setter
+    def state(self, value):
+        # Add 1 second so that when getting records are not duplicated with existing ones
+        self._state = pendulum.parse(value[self.state_pk]).add(seconds=1)
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-zendesk-talk/source_zendesk_talk/api.py:171]: it's better to have duplication than miss records. Can we remove the +1?

File: airbyte-integrations/connectors/source-zendesk-talk/source_zendesk_talk/api.py [hunk 2]
```diff
@@ -148,6 +148,49 @@ def read(self, getter: Callable) -> Iterator:
...
+        self._start_date = max(self._state, self._start_date)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._state = None
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-zendesk-talk/source_zendesk_talk/api.py:176]: just to confirm, this is being initialized with the latest state by BaseClient, right?

**Review by sherifnada — APPROVED** [review]

File: airbyte-integrations/connectors/source-zendesk-talk/source_zendesk_talk/api.py (hunk 2 above)
Comment (yevhenii-ldv) [reply airbyte-integrations/connectors/source-zendesk-talk/source_zendesk_talk/api.py:176]: yeah, of course

File: airbyte-integrations/connectors/source-zendesk-talk/source_zendesk_talk/api.py (hunk 1 above)
Comment (yevhenii-ldv) [reply airbyte-integrations/connectors/source-zendesk-talk/source_zendesk_talk/api.py]: okay

**Pull request merged**